Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -50,6 +50,8 @@ public class SiteExtensionManager : ISiteExtensionManager
         private readonly IContinuousJobsManager _continuousJobManager;
         private readonly ITriggeredJobsManager _triggeredJobManager;
 
+        private static readonly string _toBeDeletedDirectoryPath = Path.Combine(System.Environment.ExpandEnvironmentVariables(@"%HOME%\data\SiteExtensions"), "tobedeleted");
```

Comment:
why not temp folder?   then we can decide to do explicit clean-up or let DWAS remove it at site de-provisioning time.